src/ReservoirPriceOracle.sol

```diff
@@ -50,26 +50,33 @@ contract ReservoirPriceOracle is IPriceOracle, IReservoirPriceOracle, Owned(msg.
     /// @dev If `address(0)` then there is no fallback.
     address public fallbackOracle;
 
+    // TODO: Natspec must be tied to a member. In this case this comment is tied
+    // to priceDeviationThreshold. Is that correct? Maybe just do a normal (//)
+    // comment?
     /// @dev the following 4 storage variables take up 1 storage slot
 
     /// @notice percentage change greater than which, a price update may result in a reward payout of native tokens,
     /// subject to availability of rewards.
     /// 1e18 == 100%
     uint64 public priceDeviationThreshold;
 
+    // TODO: Wording is confusing. Say "This number is multiplied by the base
+    // fee to determine the reward for keepers".
     /// @notice multiples of the base fee the contract rewards the caller for updating the price when it goes
     /// beyond the `priceDeviationThreshold`
     uint64 public rewardGasAmount;
 
     /// @notice TWAP period (in seconds) for querying the oracle
     uint64 public twapPeriod;
 
+    // TODO: What's the use case for changing between price types? Should this be immutable or removed and one type be hardcoded?
     /// @notice The type of price queried and stored, possibilities as defined by `PriceType`.
     PriceType public priceType;
 
     /// @notice Designated pairs to serve as price feed for a certain token0 and token1
     mapping(address token0 => mapping(address token1 => ReservoirPair pair)) public pairs;
 
+    // TODO: Why is this needed?
```

Comment:
related to [this](https://github.com/reservoir-labs/oracle/pull/11/files#r1639693817) 

I've considered another way of doing it, which is to do an external call `IERC4626(base).asset()` but decided against it in favor of this, as external call is probably more expensive (~5K gas) compared to 1 additional SLOAD (2.1K)

Reply:
Don't you have to call the vault to access the price anyway?

Reply:
okay I just verified that, for borrowing and liquidations, we will be either querying the collateral vault to access the account's balance or to seize the collaterals. So it seems that we can call  `asset()` on the vault to access it, and it would cost minimal gas. Will implement

Reply:
implemented